### ISSUE TITLE
Configurable size (or full-screen size) on window mode

### ISSUE THREAD

**Issue by phkaeser** [phkaeser/wlmaker#98]
As reported by in #26 by @alexmyczko : "he screen size is kind of limited to some specific size (like 1/4 of my screen) on a macbook pro with retina display"

To confirm: Is that when running in window mode, or is this in a principal wayland session?

**Comment by alexmyczko** [comment]
it is startx with .xsessionrc exec wlmaker. a full session basically

**Comment by phkaeser** [comment]
Thank you for clarifying!

Please note: When running with `startx`, you'll be launching an X11 environment first, and wlmaker then runs as an application in that X11 session.

The purpose of wlmaker is to provide a Window Maker-like experience on Wayland without X11, so I recommend you rather run it from a virtual terminal without startx (Option 2 from https://github.com/phkaeser/wlmaker/blob/main/doc/RUN.md) .

That said: It'll be useful to permit configurable size (eg. adding `--width` and `--height` arguments) for running it in window mode, and should address the use-case of running as X11 session app.

**Comment by alexmyczko** [comment]
So the option 2 looks ok to me, but basically I'd probably just don't need the wrapper and can just run wlmaker. as it gets installed in /usr/bin which is in $PATH, likewise the settings for $LD_LIBRARY_PATH and $PKG_CONFIG_PATH.

I hope suppresing the options --width and --height, will assume full screen in the future. Counting that pixels is really hard...

**Comment by phkaeser** [comment]
Yes, running wlmaker from the virtual terminal (not under X11) makes it use the full output dimensions (ie. full-screen), no need for extra commandline arguments. These would be used only when running as a window (eg. under an X11 session, or from an already-running Wayland compositor).

The wrapper script is only there to set the paths when installing from source; so shouldn't be required when using the distribution's packages.

**Comment by alexmyczko** [comment]
Sorry for using it wrong…

**Comment by phkaeser** [comment]
#105 addresses this for X11 windows. 

Pending: See how to apply this when running as a window under another Wayland compositor.

**Comment by phkaeser** [comment]
* #211 finally permits `--width` and `--height` when running as window under another Wayland compositor.
* With other recent upates (#200, #201, #202, #204), `wlr-randr` (or `wdisplays`) can be used to adjust screen size, too.

**Comment by phkaeser** [comment]
Hm... noticing that while the calls are forwarded, wlroots `wayland` backend has trouble with `wlr_output_state_set_custom_mode()`. Keeping open.

**Comment by phkaeser** [comment]
(#211 was bad fix, #212 addresses this)
